com.unity.shadergraph/Editor/ShaderGraphProjectSettings.cs

```diff
@@ -0,0 +1,91 @@
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph
+{
+    [FilePath("ProjectSettings/ShaderGraphProjectSettings.asset", FilePathAttribute.Location.ProjectFolder)]
+    internal class ShaderGraphProjectSettings : ScriptableSingleton<ShaderGraphProjectSettings>
+    {
+        [SerializeField]
+        internal int customInterpolatorErrorThreshold = 32;
+        [SerializeField]
+        internal int customInterpolatorWarningThreshold = 16;
+        internal SerializedObject GetSerializedObject() { return new SerializedObject(this); }
+        internal void Save() { Save(true); }
+        private void OnDisable() { Save(); }
+    }
+
+    class ShaderGraphProjectSettingsProvider : SettingsProvider
+    {
+        private static string kCustomInterpolatorHelpBox = "These options are used to help ShaderGraph users maintain known compatibilities with target platform(s) when using Custom Interpolators.";
+        private static string kCustomInterpolatorDocumentationURL = "https://docs.unity3d.com/Packages/com.unity.shadergraph@latest/index.html?preview=1&subfolder=/manual/";
+
+        private class Styles
+        {
+            public static readonly GUIContent CustomInterpLabel = L10n.TextContent("Custom Interpolator Channel Settings", "");
+            public static readonly GUIContent CustomInterpWarnThresholdLabel = L10n.TextContent("Warning Threshold", "ShaderGraph will warn when this channel limitation is exceeded by a custom interpolator. Range must be between 8 and Error Threshold.");
+            public static readonly GUIContent CustomInterpErrorThresholdLabel = L10n.TextContent("Error Threshold", "ShaderGraph will error when this channel limitation is exceeded by a custom interpolator. Range must be between 8 and 32.");
```

Comment:
Proposed rewrite (also mentioned in another "omnibus" comment on this file):
ShaderGraph displays a warning when the user creates more custom interpolators than permitted by this setting. The number of interpolators that trigger this warning must be between 8 and Error Threshold.